### PR TITLE
refactor(cloudflare): add clarifying comments, simplify code by using `flatMap`

### PR DESCRIPTION
**Pull request by vicb**
/cc @anonrig

### CODE REVIEW

**Review by anonrig — APPROVED** [review]

File: src/presets/cloudflare.ts [hunk 1]
```diff
@@ -13,7 +14,6 @@ const cloudflareNodeCompatModules = [
   "diagnostics_channel",
   "events",
   "path",
-  "process",
```
Comment (vicb) [inline src/presets/cloudflare.ts:16]: ` process`  is in `hybridNodeCompatModules` below

File: src/presets/deno.ts [hunk 2]
```diff
@@ -56,9 +56,11 @@ const denoNodeCompatModules = [
 
 const denoPreset: Preset = {
   alias: {
-    ...Object.fromEntries(denoNodeCompatModules.map((p) => [p, `node:${p}`])),
     ...Object.fromEntries(
-      denoNodeCompatModules.map((p) => [`node:${p}`, `node:${p}`]),
+      denoNodeCompatModules.flatMap((p) => [
```
Comment (pi0) [inline src/presets/deno.ts:60]: Please limit changes to cloudflare

**Comment by pi0** [comment]
@vicb can you please update the description of exact changes?

File: src/presets/deno.ts (hunk 2 above)
Comment (vicb) [reply src/presets/deno.ts:60]: Can I ask why?

**Comment by vicb** [comment]
@pi0 could you remove the create merge commit option?

image

File: src/presets/cloudflare.ts [hunk 3]
```diff
@@ -72,10 +70,7 @@ const cloudflarePreset: Preset = {
     clearImmediate: ["unenv/runtime/node/timers/$cloudflare", "clearImmediate"],
   },
   polyfill: [],
-  external: [
-    ...cloudflareNodeCompatModules.map((p) => `${p}`),
-    ...cloudflareNodeCompatModules.map((p) => `node:${p}`),
-  ],
+  external: cloudflareNodeCompatModules.flatMap((p) => [`${p}`, `node:${p}`]),
```
Comment (pi0) [inline src/presets/cloudflare.ts:73]:
```suggestion
  external: cloudflareNodeCompatModules.flatMap((p) => [p, `node:${p}`]),
```

File: src/presets/deno.ts (hunk 2 above)
Comment (pi0) [reply src/presets/deno.ts]: it is unnecessary change.

**Pull request merged**